### PR TITLE
Add Build Trigger for 2.3-rc1 release

### PR DESCRIPTION
**Pull request by lsy323**
Similar to #5908 which added trigger for 2.2-rc1 release.

### CODE REVIEW

**Comment by zpcore** [comment]
The `v2.3.0-rc` is not out yet for pytorch, does this cause any concern if we make the change ahead of the tag creation?

**Comment by lsy323** [comment]
cc @bhavya01 who shadows this release.

**Comment by lsy323** [comment]
cc @bhavya01 who shadows this release.

File: infra/tpu-pytorch-releases/artifacts.auto.tfvars [hunk 1]
```diff
@@ -23,6 +23,60 @@ nightly_builds = [
 
 # Built on push to specific tag.
 versioned_builds = [
+  # Remove libtpu from PyPI builds
+  {
+    git_tag         = "v2.3.0-rc1"
+    package_version = "2.3.0-rc1"
+    pytorch_git_rev = "v2.3.0-rc1"
+    accelerator     = "tpu"
```
Comment (will-cromar) [inline infra/tpu-pytorch-releases/artifacts.auto.tfvars:31]: nit: can you add `python_version = "3.8"` here for completeness?

File: infra/tpu-pytorch-releases/artifacts.auto.tfvars [hunk 2]
```diff
@@ -23,6 +23,60 @@ nightly_builds = [
...
+    git_tag         = "v2.3.0-rc1"
+    package_version = "2.3.0-rc1"
+    accelerator     = "cuda"
+    cuda_version    = "12.1"
+  },
+  {
+    git_tag         = "v2.3.0-rc1"
+    package_version = "2.3.0-rc1"
+    accelerator     = "cuda"
+    cuda_version    = "12.1"
```
Comment (will-cromar) [inline infra/tpu-pytorch-releases/artifacts.auto.tfvars:77]: Can you double check that this is the CUDA version upstream is using for their release?

File: infra/tpu-pytorch-releases/artifacts.auto.tfvars [hunk 3]
```diff
@@ -23,6 +23,60 @@ nightly_builds = [
...
+    git_tag         = "v2.3.0-rc1"
+    package_version = "2.3.0-rc1+libtpu"
+    pytorch_git_rev = "v2.3.0-rc1"
+    accelerator     = "tpu"
+    python_version  = "3.10"
+    bundle_libtpu   = "1"
+  },
+  {
+    git_tag         = "v2.3.0-rc1"
+    package_version = "2.3.0-rc1"
```
Comment (will-cromar) [inline infra/tpu-pytorch-releases/artifacts.auto.tfvars:69]: The CUDA targets also need a `pytorch_git_rev`

**Review by will-cromar — APPROVED** [review]
LGTM. Heads up: the `pytorch_git_rev` for RC1 may not be `rc1`. Last time, they had some issues and their first build was with `rc4` or `rc5`. Just keep an eye on the repo when they start building

File: infra/tpu-pytorch-releases/artifacts.auto.tfvars [hunk 4]
```diff
@@ -23,6 +23,60 @@ nightly_builds = [
...
+    accelerator     = "cuda"
```
Comment (lsy323) [inline infra/tpu-pytorch-releases/artifacts.auto.tfvars:70]: Explicitly specify python version here as well

File: infra/tpu-pytorch-releases/artifacts.auto.tfvars (hunk 1 above)
Comment (lsy323) [reply infra/tpu-pytorch-releases/artifacts.auto.tfvars:31]: Sounds good, let me explicitly specify python version when we want to use the default 3.8. Added to the cuda build below as well

File: infra/tpu-pytorch-releases/artifacts.auto.tfvars (hunk 3 above)
Comment (lsy323) [reply infra/tpu-pytorch-releases/artifacts.auto.tfvars:69]: Thanks, let me add.

File: infra/tpu-pytorch-releases/artifacts.auto.tfvars (hunk 2 above)
Comment (lsy323) [reply infra/tpu-pytorch-releases/artifacts.auto.tfvars:77]: Do you know how to check this cc @vanbasten23 as well.

**Comment by lsy323** [comment]
Do not merge until upstream PyTorch has the `rc` tag created, so we can use the real `rc` number in our build.

**Comment by lsy323** [comment]
Upstream already has rc2, update the version to rc2 in tf config file.

**Review by JackCaoG — APPROVED** [review]

**Review by zpcore — APPROVED** [review]

File: infra/tpu-pytorch-releases/artifacts.auto.tfvars (hunk 2 above)
Comment (lsy323) [reply infra/tpu-pytorch-releases/artifacts.auto.tfvars:81]: Asking the PyTorch team offline, add more triggers for CUDA if necessary.

**Pull request merged**